### PR TITLE
[sentry-native] update to 0.7.0

### PR DESCRIPTION
**Pull request by AenBleidd**

### CODE REVIEW

**Comment by AenBleidd** [comment]
Ready for review

File: ports/sentry-native/fix-config-cmake.patch [hunk 1]
```diff
@@ -1,59 +1,43 @@
...
- 		include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
+ set(SENTRY_BACKEND @SENTRY_BACKEND@)
+ set(SENTRY_TRANSPORT @SENTRY_TRANSPORT@)
+ set(SENTRY_BUILD_SHARED_LIBS @SENTRY_BUILD_SHARED_LIBS@)
+@@ -8,14 +10,14 @@ set(SENTRY_LINK_PTHREAD @SENTRY_LINK_PTHREAD@)
+ if(SENTRY_BACKEND STREQUAL "crashpad" AND NOT SENTRY_BUILD_SHARED_LIBS)
+ 	include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
+ 	if(NOT MSVC AND NOT SENTRY_BUILD_SHARED_LIBS)
+-		find_package(ZLIB REQUIRED)
++		find_depenency(ZLIB REQUIRED)
```
Comment (dg0yt) [inline ports/sentry-native/fix-config-cmake.patch:18]:
```suggestion
+		find_depenency(ZLIB)
```

File: ports/sentry-native/fix-config-cmake.patch [hunk 2]
```diff
@@ -1,59 +1,43 @@
...
  	endif()
-@@ -14,7 +19,5 @@ endif()
+ endif()
+ 
+ if(SENTRY_BACKEND STREQUAL "breakpad" AND NOT SENTRY_BUILD_SHARED_LIBS)
+ 	set(SENTRY_BREAKPAD_SYSTEM @SENTRY_BREAKPAD_SYSTEM@)
+ 	if(SENTRY_BREAKPAD_SYSTEM)
+-		find_package(PkgConfig REQUIRED)
++		find_dependency(PkgConfig REQUIRED)
```
Comment (dg0yt) [inline ports/sentry-native/fix-config-cmake.patch:26]:
```suggestion
+		find_dependency(PkgConfig)
```

File: ports/sentry-native/fix-config-cmake.patch [hunk 3]
```diff
@@ -1,59 +1,43 @@
...
  include("${CMAKE_CURRENT_LIST_DIR}/sentry-targets.cmake")
-
- if(SENTRY_TRANSPORT STREQUAL "curl" AND NOT @BUILD_SHARED_LIBS@)
+ 
+ if(SENTRY_TRANSPORT STREQUAL "curl" AND (NOT @BUILD_SHARED_LIBS@ OR NOT SENTRY_BUILD_SHARED_LIBS))
 -	find_package(CURL REQUIRED)
 -	set_property(TARGET sentry::sentry APPEND
 -		PROPERTY INTERFACE_LINK_LIBRARIES ${CURL_LIBRARIES})
-+	find_dependency(CURL)
++	find_dependency(CURL REQUIRED)
```
Comment (dg0yt) [inline ports/sentry-native/fix-config-cmake.patch:37]:
```suggestion
+	find_dependency(CURL)
```

File: ports/sentry-native/fix-config-cmake.patch [hunk 4]
```diff
@@ -1,59 +1,43 @@
...
+ endif()
+ 
+ if(SENTRY_LINK_PTHREAD AND NOT SENTRY_BUILD_SHARED_LIBS)
+-	find_package(Threads REQUIRED)
++	find_dependency(Threads REQUIRED)
```
Comment (dg0yt) [inline ports/sentry-native/fix-config-cmake.patch:42]:
```suggestion
+	find_dependency(Threads)
```

File: ports/sentry-native/vcpkg.json [hunk 5]
```diff
@@ -27,6 +27,11 @@
         "Cf. https://github.com/getsentry/sentry-native#compile-time-options"
       ],
       "dependencies": [
+        {
+          "$comment": "pkgcong is used by the breakpad backend.",
+          "name": "pkgconf",
+          "platform": "!android & !ios"
```
Comment (dg0yt) [inline ports/sentry-native/vcpkg.json:33]:
```suggestion
          "name": "pkgconf",
          "platform": "!android & !ios",
          "host": true
```
But remember to run `vcpkg format-manifest ...`.

**Review by dg0yt — CHANGES_REQUESTED** [review]
`REQUIRED` is against the purpose of `find_dependency`.

**Comment by AenBleidd** [comment]
@dg0yt, done

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**